src/app/shell/admin-tools/data/provider-list/provider-list.component.ts

```diff
@@ -202,25 +408,86 @@ export class ProviderListComponent implements OnInit, OnDestroy {
               this.providerParameters
             )
           );
+        }
       });
     } else {
       const dialogRef = this.matDialog.open(ReasonModalWindowComponent, {
         data: { type: ModalConfirmationType.blockProvider }
       });
       dialogRef.afterClosed().subscribe((result: string) => {
         if (result) {
-          this.store.dispatch(new BlockProviderById({ id: provider.id, isBlocked: true, blockReason: result }, this.providerParameters));
+          this.store.dispatch(new BlockProviderById({
+            id: provider.id,
+            isBlocked: true,
+            blockReason: result
+          }, this.providerParameters));
         }
       });
     }
   }
 
+  public onResetFilters(): void {
+    if (this.searchBarFormControl.value !== '') {
+      this.searchBarFormControl.setValue('', { emitEvent: false });
+    }
+    if (this.institutionFormControl.value !== '') {
+      this.institutionFormControl.setValue('');
+    }
+    if (this.regionFormControl.value !== '') {
+      this.regionFormControl.setValue('');
+      this.areaFormControl.setValue('');
+      this.areaFormControl.disable();
+      this.store.dispatch(new ClearCodeficatorSearch());
+    }
+    if (this.areaFormControl.value !== '') {
+      this.areaFormControl.setValue('');
+      this.store.dispatch(new ClearCodeficatorSearch());
+    }
+
+    this.checkFiltersParametersForReset();
+  }
+
   ngOnDestroy(): void {
     this.destroy$.next(true);
     this.destroy$.unsubscribe();
     this.store.dispatch(new PopNavPath());
   }
 
+  private checkFiltersParametersForReset() {
+    if (this.providerParameters.searchString !== '' || this.providerParameters.size !== PaginationConstants.TABLE_ITEMS_PER_PAGE
```

Comment:
This condition looks so scary. Is there any way to optimize it?

Reply:
At least `this.providerParameters.searchString !== ''` can be just `!this.providerParameters.searchString`

And perhaps it would be good to create several variables for such conditions: `(this.isTechAdmin && (this.providerParameters.catottgId !== '' || this.providerParameters.institutionId !== ''))`

Another way is to think in reverse logic, which may require fewer conditions

Reply:
We cant make !this.providerParameters.searchString, because its not working with this condition, but at least I can optimize this condition with getters

Reply:
Creating getters may be a overkill idea if you only use them in one place. It can be just few `const`.

Reply:
I have tested this solution with the Tech Admin and the Ministry Admin, and it works great, when the filter is empty, we don't send unnecessary requests.

Reply:
Thanks for your responce and help! This is really great version of this functionality, but changed it al little bit and know it works perfect and looks much better!